op-program/client/interop/cnosolidate.go

```diff
@@ -0,0 +1,227 @@
+package interop
+
+import (
+	"fmt"
+
+	"github.com/ethereum-optimism/optimism/op-program/client/interop/types"
+	"github.com/ethereum-optimism/optimism/op-program/client/l2"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/backend/cross"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/backend/depset"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/backend/processors"
+	supervisortypes "github.com/ethereum-optimism/optimism/op-supervisor/supervisor/types"
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+)
+
+func ReceiptsToExecutingMessages(receipts ethtypes.Receipts) ([]*supervisortypes.ExecutingMessage, error) {
+	var execMsgs []*supervisortypes.ExecutingMessage
+	for _, rcpt := range receipts {
+		for _, l := range rcpt.Logs {
+			execMsg, err := processors.DecodeExecutingMessageLog(l)
+			if err != nil {
+				return nil, err
+			}
+			execMsgs = append(execMsgs, execMsg)
```

Comment:
We should skip appending `nil` to avoid confusion later.  There will be many logs which aren't executing messages and `DecodeExecutingMessageLog` returns `nil, nil` in that case.

Reply:
👍. Good catch.